pkgs/tools/networking/openssh/default.nix

```diff
@@ -19,29 +19,24 @@ in
 
   openssh_hpn = common rec {
     pname = "openssh-with-hpn";
-    version = "8.4p1";
+    version = "8.8p1";
     extraDesc = " with high performance networking patches";
 
-    src = fetchFromGitHub {
-      owner = "rapier1";
-      repo = "openssh-portable";
-      rev = "hpn-KitchenSink-${builtins.replaceStrings [ "." "p" ] [ "_" "_P" ] version}";
-      hash = "sha256-SYQPDGxZR41m4g603RaZaOYm4vCr9uZnFnZoKhruueY=";
+    src = fetchurl {
+      url = "mirror://openbsd/OpenSSH/portable/openssh-${version}.tar.gz";
+      sha256 = "1s8z6f7mi1pwsl79cqai8cr350m5lf2ifcxff57wx6mvm478k425";
     };
 
     extraPatches = [
-      ./ssh-keysign-8.4.patch
+      ./ssh-keysign-8.5.patch
 
-      # See https://github.com/openssh/openssh-portable/pull/206
-      ./ssh-copy-id-fix-eof.patch
+      ./ssh-hpn.patch
```

Comment:
This is a large patch without any comment on:

* Where exactly it is from and how one should update it (couldn't this be made a `fetchPatch`)?
* How this patch comes into existence; you write
  > using patches from the FreeBSD port `security/openssh-portable` which is regularly maintained
  but it should be explained, in comments, what the process for maintaining it if the HPN upstream repo of `rapier1` hasn't made the update yet.

Reply:
> couldn't this be made a `fetchPatch`

I never used `fetchPatch`. Updated to use it.

The original patch is from [FreeBSD ports](https://cgit.freebsd.org/ports/log/security/openssh-portable/files/extra-patch-hpn). I just adjusted it for `-p1` application.

Let me know if you need any more changes.

Thanks!

Reply:
Thanks, the `fetchpatch` makes it better.

---

Please also put the motivation, e.g. something like

> Using patches from the FreeBSD port `security/openssh-portable` instead of the upstream https://github.com/rapier1/openssh-portable/ because the former is more regularly maintained

into the comment on top of it.